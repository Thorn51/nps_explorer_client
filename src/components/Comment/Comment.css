/* Comment cards */

.comment_header {
  display: flex;
  background-color: var(--darkGunmetal);
  color: white;
  padding: 10px;
}

.author {
  width: 50%;
}

.submit_date {
  width: 50%;
  text-align: right;
}

.comment_text {
  background-color: white;
}
