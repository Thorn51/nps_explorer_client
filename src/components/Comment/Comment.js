import React from "react";
import moment from "moment";
import "./Comment.css";

export default function Comment(props) {
  return (
    <div className="comment">
      <div className="comment_header">
        <p className="author">{props.author}</p>
        <p className="submit_date">
          {moment(props.date).format("MMM Do YYYY")}
        </p>
      </div>
      <p className="comment_text">{props.comment}</p>
    </div>
  );
}
