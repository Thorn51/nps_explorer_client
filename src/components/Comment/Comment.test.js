import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Comment from "./Comment";

describe("<Comment />", () => {
  it("Renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Comment />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Renders the UI as expected", () => {
    const tree = renderer.create(<Comment />).toJSON();
    expect("tree").toMatchSnapshot();
  });
});
