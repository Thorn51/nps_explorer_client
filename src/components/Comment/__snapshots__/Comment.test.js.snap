// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment /> Renders the UI as expected 1`] = `"tree"`;
